src/runtime/hexagon/hexagon_thread_manager.cc

```diff
@@ -168,7 +180,8 @@ void HexagonThreadManager::SpawnThreads(unsigned thread_stack_size_bytes,
     next_stack_start += thread_stack_size_bytes;
 
     // create the thread
-    contexts_[i] = new ThreadContext(&pipes_[i], i);
+    contexts_[i] = new ThreadContext(&pipes_[i], i, hw_resources_.empty() ? NONE : hw_resources_[i],
+                                     hvx_.get(), htp_.get());
```

Comment:
So if we are binding resources to the thread contexts, will they no longer be available to the global thread? Because we don't have lowering support to the HTM yet, we still need these resources availble to the global thread. Do you have thoughts on how to transition the runtime from one model to the other (global thread -> stream based backed by the HTM)?

Reply:
Oh, I think I see, right now the global thread is acquiring them, and then you are passing in a pointer to the HexagonHtp or HexagonHvx. And later on you plan to acquire them (construct) directly in thread_main?

Reply:
I construct on the global thread.  I use the pointers to the object to then call Lock() on the actual threads

Reply:
I'll add a comment on the manager object construction which will hopefully make this more clear.